src/main/kotlin/com/openlattice/graph/Graph.kt

```diff
@@ -1186,6 +1190,10 @@ private val EDGE_IDS_SQL = "${EDGE_ENTITY_KEY_ID.name} = ANY(?) AND ${EDGE_ENTIT
 private val SRC_IDS_AND_ENTITY_SETS_SQL = "${SRC_ENTITY_KEY_ID.name} = ANY(?) AND ${SRC_ENTITY_SET_ID.name} = ANY(?)"
 private val DST_IDS_AND_ENTITY_SETS_SQL = "${DST_ENTITY_KEY_ID.name} = ANY(?) AND ${DST_ENTITY_SET_ID.name} = ANY(?)"
 
+const val REPARTITION_SELECTOR = "partitions[ 1 + (('x'||right(id::text,8))::bit(32)::int % array_length(partitions,1))]"
+private val REPARTITION_COLUMNS = E.columns.joinToString(",") { if (it == PARTITION) REPARTITION_SELECTOR else it.name }
+private val REPARTITION_SQL = "INSERT INTO ${E.name} SELECT $REPARTITION_COLUMNS FROM ${E.name} INNER JOIN"
```

Comment:
this is incomplete, yeah?

Reply:
correct. this is just to merge for hz-4.0